paddle/fluid/framework/data_layout_transform.cc

```diff
@@ -156,14 +157,21 @@ void TransDataLayoutFromMKLDNN(const OpKernelType& kernel_type_for_var,
 
   if (in_format != out_format) {
     void* in_data = GetDataFromTensor(in, in_type);
-    auto out_data = out->mutable_data(expected_kernel_type.place_, in.type());
+    const std::string key = platform::ReorderMKLDNNHandler::GetHash(
+        in_tz, in_format, out_format, std::to_string(in_type));
```

Comment:
Is this "key" good enough to identify two different reorder? Assume two reorder OP in one graph have exactly same dim/format/type for in and out. Then they will use same "key" value. 
We should use something unique, e.g. var name of output variable

Reply:
Despite of race condition, the key is enough, this primitive can be reused for different input/out variables since it will update internal mkldnn memory data handle(buffer ptr) for each call.  see https://github.com/PaddlePaddle/Paddle/pull/17826/files#diff-a60a81d3c97ec5a7cf7cf3d7e1aa9adbR428

But for multi-instance case, it may introduce race condition issue as Jacek is working on another Issue #17611. 
For this line, in_data was added into key to reduce potential race condition.